--- dpdk-25.03/doc/api/meson.build.orig	2025-04-20 16:11:37.174760266 +0200
+++ dpdk-25.03/doc/api/meson.build	2025-04-20 16:17:07.783616253 +0200
@@ -95,13 +95,6 @@ doxy_man_build = custom_target('doxygen-
 doc_targets += doxy_man_build
 doc_target_names += 'Doxygen_API(Manpage)'
 
-# refresh the manpage database on install
-# if DPDK manpages are installed to a staging directory, not in MANPATH, this has no effect
-mandb = find_program('mandb', required: false)
-if mandb.found() and get_option('enable_docs')
-    meson.add_install_script(mandb)
-endif
-
 # used by DTS to place its files into
 api_build_dir = meson.current_build_dir()
 
